fdbclient/MonitorLeader.actor.cpp

```diff
@@ -942,9 +1009,13 @@ ACTOR Future<MonitorLeaderInfo> monitorProxiesOneGeneration(
 			successIndex = index;
 		} else {
 			TEST(rep.getError().code() == error_code_failed_to_progress); // Coordinator cant talk to cluster controller
+			if (rep.getError().code() == error_code_coordinators_changed) {
+				throw coordinators_changed();
+			}
 			index = (index + 1) % addrs.size();
 			if (index == successIndex) {
 				wait(delay(CLIENT_KNOBS->COORDINATOR_RECONNECTION_DELAY));
+				throw coordinators_changed();
```

Comment:
Because when the client (proxy monitor) fails talking to `all` coordinators, we have to re-resolve the connection string and retry. Another option is, adding a big loop to embrace this whole function and re-resolve here, like we do in `monitorLeaderOneGeneration()`.

Reply:
Can you move this comment into the code?